services/typescript-service.ts

```diff
@@ -0,0 +1,316 @@
+				let endTime = new Date().getTime();
+				let time = (endTime - startTime) / 1000;
+
+				this.$logger.out(`${os.EOL}Success: ${time.toFixed(2)}s${os.EOL}Done without errors.`.green);
+			} else {
+				let compilerMessages = this.getCompilerMessages(compilerOutput);
+				this.logCompilerMessages(compilerMessages, compilerOutput);
+			}
+
+			return compilerOutput;
+		}).future<string>()();
+	}
+
+	private getCompilerMessages(compilerOutput: string): ITypeScriptCompilerMessages {
+		// Assumptions:
+		//   Level 1 errors = syntax errors - prevent JS emit.
+		//   Level 2 errors = semantic errors - *not* prevents JS emit.
+		//   Level 5 errors = compiler flag misuse - prevents JS emit.
+
+		let level1ErrorCount = 0,
+			level5ErrorCount = 0,
+			nonEmitPreventingWarningCount = 0;
+
+		let hasPreventEmitErrors = _.reduce(compilerOutput.split("\n"), (memo: any, errorMsg: string) => {
+			let isPreventEmitError = !!this.noEmitOnError;
+			if (errorMsg.search(/error TS1\d+:/) >= 0) {
+				level1ErrorCount += 1;
+				isPreventEmitError = true;
+			} else if (errorMsg.search(/error TS5\d+:/) >= 0) {
+				level5ErrorCount += 1;
+				isPreventEmitError = true;
+			} else if (errorMsg.search(/error TS\d+:/) >= 0) {
+				nonEmitPreventingWarningCount += 1;
+			}
+			return memo || isPreventEmitError;
+		}, false) || false;
+
+		return {
+			"level1ErrorCount": level1ErrorCount,
+			"level5ErrorCount": level5ErrorCount,
+			"nonEmitPreventingWarningCount": nonEmitPreventingWarningCount,
+			"hasPreventEmitErrors": hasPreventEmitErrors
+		};
+	}
+
+	private logCompilerMessages(compilerMessages: ITypeScriptCompilerMessages, errorMessage: string): void {
+		let level1ErrorCount = compilerMessages.level1ErrorCount,
+			level5ErrorCount = compilerMessages.level5ErrorCount,
+			nonEmitPreventingWarningCount = compilerMessages.nonEmitPreventingWarningCount,
+			hasPreventEmitErrors = compilerMessages.hasPreventEmitErrors;
+
+		if (level1ErrorCount + level5ErrorCount + nonEmitPreventingWarningCount > 0) {
+			let colorizedMessage = (level1ErrorCount + level5ErrorCount > 0) ? ">>>".red : ">>>".green;
+			this.$logger.out(colorizedMessage);
+
+			let errorTitle = "";
+			if (level5ErrorCount > 0) {
+				errorTitle += this.composeErrorTitle(level5ErrorCount, "compiler flag error");
+			}
+			if (level1ErrorCount > 0) {
+				errorTitle += this.composeErrorTitle(level1ErrorCount, "syntax error");
+			}
+			if (nonEmitPreventingWarningCount > 0) {
+				if (!level1ErrorCount && !level5ErrorCount && this.noEmitOnError) {
+					errorTitle += this.composeErrorTitle(nonEmitPreventingWarningCount, "non-emit-preventing type errors, but output is not generated as noEmitOnError option is true.");
+				} else {
+					errorTitle += this.composeErrorTitle(nonEmitPreventingWarningCount, "non-emit-preventing type warning");
+				}
+			}
+
+			if (hasPreventEmitErrors) {
+				process.stderr.write(os.EOL + <any>errorTitle);
```

Comment:
I think you can use `$errors.failWithoutHelp` here instead of writing to `stderr` and `process.exit`-ing.